ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/CorsAllowedOriginsProperty.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core;
+
+import java.util.AbstractList;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.StringJoiner;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+import javax.annotation.Nonnull;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+
+public class CorsAllowedOriginsProperty extends AbstractList<String> {
+
+  private final List<String> domains = new ArrayList<>();
+
+  public CorsAllowedOriginsProperty() {}
+
+  @Override
+  @Nonnull
+  public Iterator<String> iterator() {
+    if (domains.size() == 1 && domains.get(0).equals("none")) {
+      return Collections.emptyIterator();
+    } else {
+      return domains.iterator();
+    }
+  }
+
+  @Override
+  public int size() {
+    return domains.size();
+  }
+
+  @Override
+  public boolean add(final String string) {
+    return addAll(Collections.singleton(string));
+  }
+
+  @Override
+  public String get(final int index) {
+    return domains.get(index);
+  }
+
+  @Override
+  public boolean addAll(final Collection<? extends String> collection) {
+    final int initialSize = domains.size();
+    for (final String string : collection) {
+      if (Strings.isNullOrEmpty(string)) {
+        throw new IllegalArgumentException("Domain cannot be empty string or null string.");
+      }
+      for (final String s : Splitter.onPattern("\\s*,+\\s*").split(string)) {
+        if ("all".equals(s)) {
+          domains.add("*");
+        } else {
+          domains.add(s);
+        }
+      }
+    }
+
+    if (domains.contains("none")) {
```

Comment:
Can tests be added for the none and all cases, these look like special conditions that should be tested.

Reply:
that is some nasty code - what if the length is 2?

Reply:
I think it fails. Looking at the tests in Besu this is meant to fail if none or * is combined with any other domain values.